src/components/modal/profile/index.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useContext } from 'react';
+import Backdrop from '@mui/material/Backdrop';
+import Modal from '@mui/material/Modal';
+import Fade from '@mui/material/Fade';
+import Typography from '@mui/material/Typography';
+import Avatar from '@mui/material/Avatar';
+import { LoadingButton } from '@mui/lab';
+import avatar from '../../../static/images/john-doe.png';
+import { Box } from '@mui/material';
+import { InteractiveComponentsContext } from 'contexts/interactiveComponentsContext';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import { useTheme } from '@mui/material/styles';
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  alignItems: 'center',
+  overflow: 'hidden',
+};
+
+export default function ProfileModal() {
+  const { isProfileModalOpened, closeProfileModal } = useContext(InteractiveComponentsContext);
+  const theme = useTheme();
+
+  function logout() {
+    closeProfileModal();
+    //TODO: add logout functionality
+  }
+
+  function goToProfilePage() {
+    closeProfileModal();
+    //TODO: route to profile page
+  }
+
+  return (
+    <Modal
```

Comment:
We should have a generic modal that in terms of style is always centered on the middle of the screen and has a backdrop and no fixed width/height. The width/height should be determined by the component inside of the modal.

Reply:
Have create a generic modal for general use in src/components/modal/index.tsx. This previously mentioned modal is then imported and used in the **profile modal** - src/components/modal/profile/index.tsx.